---
categories:
- AIfastConnectome
title: JEEK - Fast and Scalable Joint Estimator for Integrating Additional Knowledge in Learning Multiple Related Sparse Gaussian Graphical Models
tag:
- multi-Graph-generative
- ExtraPrior
- Graph-generative
---


## Tool JEEK: A Fast and Scalable Joint Estimator for Integrating Additional Knowledge in Learning Multiple Related Sparse Gaussian Graphical Models

### Paper: Most updated version at [HERE](https://github.com/QData/JEEK/blob/master/icml18-JEEKarxivUpdated.pdf) | Previous version: [@Arxiv](https://arxiv.org/abs/1806.00548) | 
### URL at  [2018 ICML](http://proceedings.mlr.press/v80/wang18f.html)

### TalkSlide: [URL](https://github.com/QData/JEEK/blob/master/JEEK-ICML18talk.pdf)

### R package: [URL](https://cran.r-project.org/web/packages/jeek/index.html)

### GitRepo for R package: [URL](https://github.com/QData/JEEK)


```R
install.packages("jeek")
library(jeek)
demo(jeek)
```


### Abstract
We consider the problem of including additional knowledge in estimating sparse Gaussian graphical models (sGGMs) from aggregated samples, arising often in bioinformatics and neuroimaging applications. Previous joint sGGM estimators either fail to use existing knowledge or cannot scale-up to many tasks (large $K$) under a high-dimensional (large $p$) situation.  In this paper, we propose a  novel \underline{J}oint \underline{E}lementary \underline{E}stimator incorporating additional \underline{K}nowledge (JEEK) to infer multiple related sparse Gaussian Graphical models from large-scale heterogeneous data. Using domain knowledge as weights, we design a novel hybrid norm as the minimization objective to enforce the superposition of two weighted sparsity constraints, one on the shared interactions and the other on the task-specific structural patterns. This enables JEEK to elegantly consider various forms of existing knowledge based on the domain at hand and avoid the need to design knowledge-specific optimization. JEEK is solved through a fast and entry-wise parallelizable solution that largely improves the computational efficiency of the state-of-the-art  $O(p^5K^4)$ to $O(p^2K^4)$. We conduct a rigorous statistical analysis showing that JEEK achieves the same  convergence rate $O(\log(Kp)/n_{tot})$ as the state-of-the-art estimators that are much harder to compute. 
Empirically, on multiple synthetic datasets and two real-world data, JEEK outperforms the speed of the state-of-arts significantly while achieving the same level of prediction accuracy.


### About Adding Additional Knowledge 

One significant caveat of state-of-the-art joint sGGM estimators is the fact that little attention has been paid to incorporating existing knowledge of the nodes or knowledge of the relationships among nodes in the models. 
In addition to the samples themselves, additional information is widely available in real-world applications. In fact, incorporating the knowledge is of great scientific interest. A prime example is when estimating the functional brain connectivity networks among brain regions based on fMRI samples, the spatial position of the regions are readily available. Neuroscientists have gathered considerable knowledge regarding the spatial and anatomical evidence underlying brain connectivity (e.g., short edges and certain anatomical regions are more likely to be connected \cite{watts1998collective}). Another important example is the problem of identifying gene-gene interactions from patients' gene expression profiles across multiple cancer types. Learning the statistical dependencies among genes from such heterogeneous datasets can help to understand how such dependencies vary from normal to abnormal and help to discover contributing markers that influence or cause the diseases. Besides the patient samples, state-of-the-art bio-databases like HPRD \cite{prasad2009human} have collected a significant amount of information about direct physical interactions among corresponding proteins, regulatory gene pairs or signaling relationships collected from high-qualify bio-experiments. 



Although being strong evidence of structural patterns we aim to discover, this type of information has rarely been considered in the joint sGGM formulation of such samples. This paper aims to fill this gap by adding additional knowledge most effectively into scalable and fast joint sGGM estimations.


The proposed JEEK estimator provides the flexibility of using ($K+1$) different weight matrices representing the extra knowledge. We try to showcase a few possible designs of the weight matrices, including (but not limited to): 

- Spatial or anatomy knowledge about brain regions;
- Knowledge of known co-hub nodes or perturbed  nodes; 
- Known group information about nodes, such as genes belonging to the same biological pathway or cellular location; 
- Using existing known edges as the knowledge, like the known protein interaction databases for discovering gene networks (a semi-supervised setting for such estimations).

We sincerely believe the scalability and flexibility provided by JEEK can make structure learning of joint sGGM feasible in many real-world tasks. 


##### an example W for how to add known group sparity
![jeekGroup]({{ site.baseurl }}pic/group.png){:class="img-responsive"}

##### an example W for how to add known group interactions
![jeekGroup]({{ site.baseurl }}pic/group2.png){:class="img-responsive"}

##### an example W for how to add known hub node
![jeekGroup]({{ site.baseurl }}pic/hub.png){:class="img-responsive"}

##### an example W for how to add known perturbed-hub node
![jeekGroup]({{ site.baseurl }}pic/hub2per.png){:class="img-responsive"}



### Citations

```latex
@conference{wang2018jeek,
  Author = {Wang, Beilun and Sekhon, Arshdeep and Qi, Yanjun},
  Booktitle = {Proceedings of The 35th International Conference on Machine Learning (ICML)},
  Title = {A Fast and Scalable Joint Estimator for Integrating Additional Knowledge in Learning Multiple Related Sparse Gaussian Graphical Models},
  Year = {2018}}
}
```


### Support or Contact

Having trouble with our tools? Please [contact Beilun](mailto:bw4mw@virginia.edu) and weâ€™ll help you sort it out.
